pkg/reconciler/revision/resources/deploy.go

```diff
@@ -36,49 +36,88 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
+
+	apiconfig "knative.dev/serving/pkg/apis/config"
 )
 
-//nolint:gosec // Filepath, not hardcoded credentials
-const concurrencyStateTokenVolumeMountPath = "/var/run/secrets/tokens"
-const concurrencyStateTokenName = "state-token"
+// VolumeNames
+//nolint:gosec // VolumeName, not hardcoded credentials
+const (
+	tokenVolumeName   = "knative-token-volume"
+	podInfoVolumeName = "pod-info"
+	certVolumeName    = "server-certs"
+	logVolumeName     = "knative-var-log"
+
+	PodInfoFeatureKey     = "features.knative.dev/podspec-podinfo"
+	PodInfoFeatureEnabled = "enabled"
+)
 
 var (
+	// Volumes
 	varLogVolume = corev1.Volume{
-		Name: "knative-var-log",
+		Name: logVolumeName,
 		VolumeSource: corev1.VolumeSource{
 			EmptyDir: &corev1.EmptyDirVolumeSource{},
 		},
 	}
 
-	varLogVolumeMount = corev1.VolumeMount{
-		Name:        varLogVolume.Name,
-		MountPath:   "/var/log",
-		SubPathExpr: "$(K_INTERNAL_POD_NAMESPACE)_$(K_INTERNAL_POD_NAME)_",
-	}
-
 	varTokenVolume = corev1.Volume{
-		Name: "knative-token-volume",
+		Name: tokenVolumeName,
 		VolumeSource: corev1.VolumeSource{
 			Projected: &corev1.ProjectedVolumeSource{
 				Sources: []corev1.VolumeProjection{{
 					ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
 						ExpirationSeconds: ptr.Int64(600),
-						Path:              concurrencyStateTokenName,
+						Path:              queue.ConcurrencyStateTokenFilename,
```

Comment:
I am ok either way, my thought process is that I prefer to see an alignment between the way the 4 volumes are defined and allow all Mount points and Filenames to be accessible in queue. Specifically for podInfo, I can see a reason why we will need it in sharedmain moving forward (to provide each option its own config rather than having all features parse podinfo/annotations file).